Add back tests for RetrieveChat

thinkall opened the pull request:
<!-- Thank you for your contribution! Please review https://microsoft.github.io/autogen/docs/Contribute before opening a pull request. -->

<!-- Please add a reviewer to the assignee section when you create a PR. If you don't have the access to it, we will shortly find a reviewer and assign them to your PR. -->

## Why are these changes needed?

Add back tests for RetrieveChat.  Related to #393 .

## Related issue number

Closes #451 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

rickyloynd-microsoft reviewed (approved):
My experience with pytest workflows is quite shallow, so I don't see any problems with this PR. But I'll definitely follow this pattern for the upcoming PR to restore tests for TeachableAgent. Thanks!

sonichi reviewed (approved):
Looks ready for merging. What's the plan for running tests which require openai?

thinkall commented:
> Looks ready for merging. What's the plan for running tests which require openai?

What if we split the tests into optional and required categories as well? For the required OpenAI tests, they will automatically run when added to the merge queue. For the optional OpenAI tests, you can approve the tests when necessary in a specific case.

sonichi commented:
> > Looks ready for merging. What's the plan for running tests which require openai?
> 
> What if we split the tests into optional and required categories as well? For the required OpenAI tests, they will automatically run when added to the merge queue. For the optional OpenAI tests, you can approve the tests when necessary in a specific case.

All OpenAI tests need approval. Separation is good, thought not necessarily in different workflows. For example, we can run core test in early steps and optional test in later steps. I'm not sure and would like to hear others' thoughts.

rickyloynd-microsoft commented:
> > > Looks ready for merging. What's the plan for running tests which require openai?
> > 
> > 
> > What if we split the tests into optional and required categories as well? For the required OpenAI tests, they will automatically run when added to the merge queue. For the optional OpenAI tests, you can approve the tests when necessary in a specific case.
> 
> All OpenAI tests need approval. Separation is good, thought not necessarily in different workflows. For example, we can run core test in early steps and optional test in later steps. I'm not sure and would like to hear others' thoughts.

To keep the core tests completely unblocked by contributed components, I would suggest making the high-level split between core and contrib. Then core tests (as now) would be split into its initial tests, followed (after approval) by its openai tests. Similarly, the contrib tests (as @thinkall suggested) should probably be split into its initial tests, followed (after approval) by its openai tests. So, a contributed component would need to pass all 4 sets of tests before being added to the merge queue. Am I overlooking something? Maybe this was already suggested.

rickyloynd-microsoft commented:
> > > > Looks ready for merging. What's the plan for running tests which require openai?
> > > 
> > > 
> > > What if we split the tests into optional and required categories as well? For the required OpenAI tests, they will automatically run when added to the merge queue. For the optional OpenAI tests, you can approve the tests when necessary in a specific case.
> > 
> > 
> > All OpenAI tests need approval. Separation is good, thought not necessarily in different workflows. For example, we can run core test in early steps and optional test in later steps. I'm not sure and would like to hear others' thoughts.
> 
> To keep the core tests completely unblocked by contributed components, I would suggest making the high-level split between core and contrib. Then core tests (as now) would be split into its initial tests, followed (after approval) by its openai tests. Similarly, the contrib tests (as @thinkall suggested) should probably be split into its initial tests, followed (after approval) by its openai tests. So, a contributed component would need to pass all 4 sets of tests before being added to the merge queue. Am I overlooking something? Maybe this was already suggested.

Actually, my last post looks completely backwards to me now! How about the following?

**Test sequence for core components**
core non-openai tests
approval
core openai tests

**Test sequence for contrib components**
core non-openai tests
contrib non-openai tests
approval
core openai tests
contrib openai tests

sonichi commented:
@rickyloynd-microsoft sounds good to me.

sonichi commented:
@thinkall could you create a task list in #480 to add a task about the test requiring openai? Assuming you plan to do it in a separate PR.

thinkall commented:
> @thinkall could you create a task list in #480 to add a task about the test requiring openai? Assuming you plan to do it in a separate PR.

Added openai tests, separated contrib-openai tests into a new workflow because core and contrib tests will be triggered by different file changes.

Renamed the first job in contrib tests to RetrieveChatTest, maybe others like math and teachable can be added in different jobs in the workflow, so each test will run in its own env.

sonichi commented on .github/workflows/build.yml:
```diff
@@ -49,7 +49,7 @@ jobs:
         if: matrix.python-version == '3.10'
         run: |
-          pip install -e.[mathchat,test]
+          pip install -e .[test]
```
One existing test needs [mathchat].

sonichi commented:
> > @thinkall could you create a task list in #480 to add a task about the test requiring openai? Assuming you plan to do it in a separate PR.
> 
> Added openai tests, separated contrib-openai tests into a new workflow because core and contrib tests will be triggered by different file changes.
> 
> Renamed the first job in contrib tests to RetrieveChatTest, maybe others like math and teachable can be added in different jobs in the workflow, so each test will run in its own env.
> 
> @sonichi , @rickyloynd-microsoft

Thanks. @kevin666aa @BeibinLi for awareness.

sonichi commented on .github/workflows/contrib-openai.yml:
```diff
@@ -0,0 +1,60 @@
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.9", "3.10", "3.11"]
```
It looks that only py3.10 enables the test. Do we need the multiple versions here?

thinkall replied:
Removed.

thinkall commented:
> Should this be added to `build.yml` to separate the tests?
> 
> ```
>     paths-ignore:
>       - 'test/agentchat/contrib/**'
> ```
> 
> Currently `build.yml` is not run because this branch is merging in to `dev`

build.yml will run once this PR is merged. I think we can keep it as any changes should pass the core tests.
